crates/vm/levm/src/vm.rs

```diff
@@ -589,14 +589,13 @@ impl VM {
             .checked_mul(BLOB_GAS_PER_BLOB)
             .unwrap_or_default();
 
-        let blob_gas_cost = self
-            .env
-            .tx_max_fee_per_blob_gas
-            .unwrap_or_default()
-            .checked_mul(blob_gas_used)
+        let base_fee_per_blob_gas = self.get_base_fee_per_blob_gas()?;
+
+        let blob_fee = blob_gas_used
+            .checked_mul(base_fee_per_blob_gas)
             .ok_or(InternalError::UndefinedState(1))?;
```

Comment:
This seemed to be the issue. The [EIP-4844](https://eips.ethereum.org/EIPS/eip-4844#gas-accounting) describes how to the gas for the blob hashes should be computed.

Before this patch, it used the max fee per blob gas to calculate the value. However, that makes me wonder where the max fee per blob gas is used.

Regardless, more tests seem to pass(?